### ISSUE TITLE
Detect infinite loops in XSON to avoid stack overflows.

### ISSUE THREAD

**Issue by chrhol** [Starcounter/Starcounter#4085]
There are some places in Starcounter.XSON that might end up in an infinite loop which will lead to stackoverflow caused by the following: 

- a json is connected to a parent (somewhere in the tree) that is a child of the json.
- a merge where one sibling is a parent (somewhere in the tree) to the other sibling

This cases should be detected and handled instead of ending up in a stackoverflow. Hopefully enough information can be provided  to be able to figure out whats caused the infinite loop. 

The sample below will end up in a stackoverflow. Of course in this sample it's really easy to see what the problem is. Most of the time when something like this happens it's not so obvious.

**Test.json**
```javascript
{
  "Page": {}
}
```

**Program.cs**
```csharp
class Program {
    static void Main() {
          Handle.GET("/test", () => {
               Test x = new Test();
               Test y = new Test();
               Test z = new Test();

               x.Page = y;
               y.Page = z;
               z.Page = x;

               return x;
          });
    }
}

**Comment by chrhol** [comment]
I got stuck on other issues, and still haven't found the time to finish this one. I will remove the high prio on it I think.

**Comment by mtornwall** [comment]
Can this actually occur absent developer error?

**Comment by chrhol** [comment]
> Can this actually occur absent developer error?

No, I don't think so. All cases when it's happened it's been an mistake or error from the app developer. The problem is that it's so hard to track down when it happens and take a lot of time to find.
